backend/dataall/modules/s3_datasets/api/dataset/resolvers.py

```diff
@@ -167,6 +173,29 @@ def validate_creation_request(data):
         ConfidentialityClassification.validate_confidentiality_level(data.get('confidentiality', ''))
         if len(data['label']) > 52:
             raise InvalidInput('Dataset name', data['label'], 'less than 52 characters')
+        if data.get('enableExpiration', False):
+            RequestValidator.validate_share_expiration_request(data)
+
+    @staticmethod
+    def validate_share_expiration_request(data):
+        if data.get('expiryMinDuration') < 0 or data.get('expiryMaxDuration') < 0:
```

Comment:
This expression will raise an error if `data.get('expiryMinDuration') is None`.

Reply:
Thanks for pointing this out. I was thinking its good to raise an error in case an invalid input is given. But I can see that error which will get raised here would be unclear. I will add explicit type checking and raise a custom error in case a valid is not present